diff --git a/src/main/index.ts b/src/main/index.ts
index 4cc2e0db..d119ba17 100644
--- a/src/main/index.ts
+++ b/src/main/index.ts
@@ -106,7 +106,7 @@ if (IS_VESKTOP || !IS_VANILLA) {
             }
         };

-        session.defaultSession.webRequest.onHeadersReceived(({ responseHeaders, resourceType }, cb) => {
+        session.defaultSession.webRequest.onHeadersReceived(({ responseHeaders, resourceType, url }, cb) => {
             if (responseHeaders) {
                 if (resourceType === "mainFrame")
                     patchCsp(responseHeaders);
@@ -118,6 +118,13 @@ if (IS_VESKTOP || !IS_VANILLA) {
                     if (header)
                         responseHeaders[header] = ["text/css"];
                 }
+                // Fix CORS
+                if (url.startsWith("https://cdn.discordapp.com/attachments/")) {
+                    const header = findHeader(responseHeaders, "access-control-allow-origin");
+                    if (header) return;
+                    responseHeaders["Access-Control-Allow-Origin"] = ["*"];
+                    responseHeaders["Access-Control-Allow-Methods"] = ["GET, POST, OPTIONS"];
+                }
             }

             cb({ cancel: false, responseHeaders });
